pkg/model/field.go

```diff
@@ -178,6 +178,21 @@ func (f *Field) ReferencedResourceNamePlural() string {
 	return referencedResourceName
 }
 
+// ReferenceFieldPath returns the fieldPath for the corresponding
+// Reference field. It replaces the fieldName with ReferenceFieldName
+// at the end of fieldPath
+func (f *Field) ReferenceFieldPath() string {
+	fieldPathPrefix := strings.TrimSuffix(f.Path, f.Names.Camel)
+	return fmt.Sprintf("%s%s", fieldPathPrefix, f.GetReferenceFieldName().Camel)
+}
+
+// CamelCasedFieldPath replaces the period in fieldPath with
+// underscore. This method is useful for generating go method
+// name from the fieldPath.
+func (f *Field) CamelCasedFieldPath() string {
+	return strings.ReplaceAll(f.Path, ".", "_")
+}
```

Comment:
I renamed the method but keeping it on *model.Field instead of fieldPath because I need to use this method inside references.go.tpl template. 
The suggested refactor can be used inside go template if model.field.Path return fieldPath.Path instead of string but I believe that can be tackled in future PR if i need to use this method again.